packages/components/src/SegmentedControl/SegmentedControl.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { type PropsWithChildren, useId, useRef } from "react";
+import { SegmentedControlProvider } from "./SegmentedControlProvider";
+import { SegmentedControlOption } from "./SegmentedControlOption";
+import { SegmentedControlBase } from "./SegmentedControlBase";
+
+interface SegmentedControlProps<T> extends PropsWithChildren {
+  /**
+   * The currently selected option
+   */
+  readonly selectedValue?: T;
+
+  /**
+   * A callback function that is called whenever the selected option changes
+   */
+  readonly onSelectValue?: (value: T) => void;
+
+  /**
+   * The default option to be selected initially
+   */
+  readonly defaultValue?: T;
+
+  /**
+   * A unique name for the SegmentedControl, that links the group of
+   * options together.
+   *
+   * @default useId()
+   */
+  readonly name?: string;
```

Comment:
@Aiden-Brine I played around with having the `name` here or in the option, but decided to have `name` in the parent and pass it down through the provider. This centralizes management of the `name` prop and the user doesn't have to set `name` on each option, which can be a headache/redundant/leave room for error if `name` isn't consistent between options, like a typo or something.

Reply:
I agree that this is a better location for it 👍